src/store/modules/canvas.ts

```diff
@@ -0,0 +1,93 @@
+import { ActionContext, Module } from 'vuex'
+import { CanvasElement } from '@/types/Canvas'
+import uuid from 'uuid'
+
+export enum CanvasMutation {
+  SET_CANVAS_ELEMENT = 'SET_CANVAS_ELEMENT',
+  ADD_CANVAS_ELEMENT = 'ADD_CANVAS_ELEMENT',
+  REMOVE_CANVAS_ELEMENT = 'REMOVE_CANVAS_ELEMENT',
+  REDO_CANVAS_ELEMENT = 'REDO_CANVAS_ELEMENT'
+}
+
+export enum CanvasAction {
+  SET_CANVAS_ELEMENT = 'setCanvasElement',
+  ADD_CANVAS_ELEMENT = 'addCanvasElement',
+  REMOVE_CANVAS_ELEMENT = 'removeCanvasElement',
+  REDO_CANVAS_ELEMENT = 'redoCanvasElement'
+}
+
+export enum CanvasGetters {
+  CANVAS_ELEMENTS = 'canvasElements',
+  CANVAS_ELEMENTS_HISTORY = 'canvasElementsHistory'
+}
+
+interface CanvasState {
+  canvasElements: CanvasElement[];
+  canvasElementsHistory: CanvasElement[];
+}
+
+type CursorActionContext = ActionContext<CanvasState, {}>
+
+const CursorModule: Module<CanvasState, {}> = {
+  namespaced: true,
+  state () {
+    return {
+      canvasElements: [],
+      canvasElementsHistory: []
+    }
+  },
+  getters: {
+    [CanvasGetters.CANVAS_ELEMENTS]: state => state.canvasElements,
+    [CanvasGetters.CANVAS_ELEMENTS_HISTORY]: state => state.canvasElementsHistory
+  },
+  mutations: {
+    [CanvasMutation.SET_CANVAS_ELEMENT] (state: CanvasState, payload: CanvasElement[]) {
+      state.canvasElements = [...payload]
+    },
+    [CanvasMutation.ADD_CANVAS_ELEMENT] (state: CanvasState, payload: CanvasElement) {
+      payload.id = uuid()
+      payload.jti = '11111'
+      state.canvasElements = [...state.canvasElements, payload as CanvasElement]
+    },
+    [CanvasMutation.REMOVE_CANVAS_ELEMENT] (state: CanvasState, id: string) {
+      let foundElementIndex = -1
+      const foundElement = state.canvasElements.find((canvasElement: CanvasElement, index: number) => {
+        foundElementIndex = index
+        return canvasElement.id === id
+      })
+      if (foundElementIndex && foundElement) {
+        state.canvasElementsHistory = [...state.canvasElementsHistory, foundElement]
```

Comment:
use concat instead of reassigning the entire array

`state.canvasElementsHistory,concat(foundElement)`

Reply:
Is this for performance reasons?
FYI. https://stackoverflow.com/a/48865797/8490159
Happy for me to change it to Concat?